src/Notifications/RefreshToken/AbstractRefreshTokenNotification.php

```diff
@@ -41,4 +43,34 @@ abstract class AbstractRefreshTokenNotification extends AbstractNotification
     {
         return true;
     }
+
+    /**
+     * @var array
+     */
+    protected $tags = ['refresh_token'];
+
+    public $user_name;
+
+    public $image;
+
+    public $main_character;
+
+    public $corporation;
+
+    public $refresh_token;
+
+    public function __construct(RefreshToken $refresh_token)
+    {
+        parent::__construct();
+
+        $this->refresh_token = $refresh_token;
+        $this->user_name = $refresh_token->user->name;
+        $this->image = 'https://imageserver.eveonline.com/Character/' . $refresh_token->character_id . '_128.jpg';
+        $this->main_character = $this->getMainCharacter($refresh_token->user->group)->name;
```

Comment:
ensure you'll not have any `null` exception. This is a constructor !

Reply:
this method is a reuse from seat-groups. (you have actually wrote it):

https://github.com/herpaderpaldent/seat-notifications/blob/0af667689342ea8b432e564af4370cdb86ebe347/src/Notifications/AbstractNotification.php#L110-L122